### ISSUE TITLE
Test-Rule commandlet to check objects compliance against Sitecore rules

### ISSUE THREAD

**Issue by AdamNaj** [SitecorePowerShell/Console#327]
```PowerShell
Test-Rule [-Rule <string>] [-InputObject <psobject>] [-RuleDatabase <string>]
```
- ```-Rule``` - rule XML either created by Read-Variable cmdlet or read from a field.
- ```-InputObject``` - object to be evaluated
- ```-RuleDatabase``` - database containing the rule definitions

Example - this script finds all Items of template "Sample Templates" under ```master:\content``` item:

```powershell
$rule = '<ruleset>
  <rule uid="{9CF02118-F189-49C4-9F2B-6698D64ACF23}">
    <conditions>
      <condition id="{5DE6D53E-EA62-4D17-8BDD-FEBCD80AC07B}" uid="6C620AC228E84465BF237924D9023D77" templateid="{76036F5E-CBCE-46D1-AF0A-4143F9B557AA}" />
    </conditions>
  </rule>
</ruleset>'


Get-ChildItem master:\content\ -Recurse | ? { Test-Rule -InputObject $_ -Rule $rule -RuleDatabase master}
```

**Comment by michaellwest** [comment]
Can you get the rule database from the item $_?

**Comment by AdamNaj** [comment]
I do already. If you don't specify the parameter and ```$_``` is ```Sitecore.Data.Item``` I'll take the Database from there, otherwise it will take the Database SPE config file ```configuration/sitecore/powershell/workingDatabase/rules```

```xml
<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/">
  <sitecore>
    <powershell>
      <workingDatabase>
        <rules>master</rules>
      </workingDatabase>
    </powershell>
  </sitecore>
</configuration>
```

**Comment by michaellwest** [comment]
Great. Glad you got it covered.

**Comment by AdamNaj** [comment]
Merged by implementing Issue #350
